### ISSUE TITLE
Formalize TrainableStrategy protocol and extract shared metric utilities

### ISSUE THREAD

**Issue by Acusick1** [Acusick1/odds#139]
## Goal

The XGBoost and LSTM training paths share the same `train_from_config` / `train_with_cv` interface by convention only — enforced via a `TypeAlias` union, not a protocol. Metric computation (MSE, MAE, R²) is manually reimplemented 4 times across the two strategy files. Formalizing the interface and extracting shared utilities reduces duplication and makes adding new model types straightforward.

## Scope

- Define a `TrainableStrategy` `Protocol` with `train_from_config()` and `train_with_cv()` signatures, replacing the `TypeAlias` union in `cross_validation.py`
- Extract `compute_regression_metrics(y_true, y_pred) -> dict` utility (returns `mse`, `mae`, `r2`) to replace the 4 hand-rolled copies
- Deduplicate `LSTMLineMovementStrategy` — it has two separate epoch loops in `train()` and `train_from_config()` doing the same thing; consolidate into one
- Consolidate `train_with_cv()` — both strategies have identical implementations (call `run_cv`, train final, merge dicts); consider moving to a shared function or mixin
- Assess whether `FeatureExtractor` ABC adds value now that only `TabularFeatureExtractor` and `SequenceFeatureExtractor` remain (post #138); if not, flatten the hierarchy

## Location

- `packages/odds-analytics/odds_analytics/training/cross_validation.py` — `TrainableStrategy` definition
- `packages/odds-analytics/odds_analytics/training/utils.py` — new metric utility
- `packages/odds-analytics/odds_analytics/xgboost_line_movement.py`
- `packages/odds-analytics/odds_analytics/lstm_line_movement.py`
- `packages/odds-analytics/odds_analytics/feature_extraction.py` — ABC assessment

## Implementation approach

```python
# training/protocols.py (or inline in cross_validation.py)
class TrainableStrategy(Protocol):
    def train_from_config(
        self,
        config: MLTrainingConfig,
        X_train: np.ndarray,
        y_train: np.ndarray,
        feature_names: list[str],
        X_val: np.ndarray | None = None,
        y_val: np.ndarray | None = None,
        tracker: ExperimentTracker | None = None,
        trial: Any | None = None,
    ) -> dict[str, Any]: ...

    def train_with_cv(
        self,
        config: MLTrainingConfig,
        X: np.ndarray,
        y: np.ndarray,
        feature_names: list[str],
        X_test: np.ndarray | None = None,
        y_test: np.ndarray | None = None,
        event_ids: np.ndarray | None = None,
    ) -> tuple[dict[str, Any], CVResult]: ...
```

```python
# training/utils.py
def compute_regression_metrics(y_true: np.ndarray, y_pred: np.ndarray) -> dict[str, float]:
    """Return dict with keys: mse, mae, r2."""
```

## Constraints

- Depends on #138 (less code to refactor once dead paths are gone)
- No new dependencies
- Protocol should be runtime-checkable if practical

## Open questions

- Should `train_with_cv` be a standalone function that wraps any `TrainableStrategy`, rather than a method each strategy must implement? Both current implementations are identical.
- Is the `FeatureExtractor` ABC still justified with only two concrete subclasses, or should `TabularFeatureExtractor` and `SequenceFeatureExtractor` just be standalone classes?

## Success criteria

- [ ] `TrainableStrategy` is a `Protocol`, not a `TypeAlias`
- [ ] `compute_regression_metrics()` exists in `training/utils.py` and is used by both strategies
- [ ] No hand-rolled MSE/MAE/R² computation remains in strategy files
- [ ] `LSTMLineMovementStrategy` has a single training loop, not two
- [ ] `train_with_cv` logic is not duplicated between strategies
- [ ] All existing tests pass
- [ ] `ruff check` and `ruff format` clean
